incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -191,6 +191,29 @@ var _ = Describe("Secret", func() {
 		Eventually(objectInheritedFrom(ctx, "Role", barName, "bar-role")).Should(Equal(fooName))
 	})
 
+	It("should overwrite conflicting source with the top source that can propagate", func() {
+		// Create a 'baz-role' in 'foo' that cannot propagate because of the finalizer.
+		makeObject(ctx, "Role", fooName, "baz-role")
+		Eventually(hasObject(ctx, "Role", fooName, "baz-role")).Should(BeTrue())
+		setFinalizer(ctx, fooName, "baz-role", true)
```

Comment:
Added line 197 before `setFinalizer`. The previous flaky "baz-role" not found error should be fixed.